horovod/common/operations.cc

```diff
@@ -1811,10 +1811,14 @@ void BackgroundThreadLoop(HorovodGlobalState& state) {
   // Signal that initialization is completed.
   state.initialization_done = true;
 
+  LOG(INFO, rank) << "Horovod Initialized";
```

Comment:
It gets printed immediately (before the background thread is terminated) because the object is created without a handle, so it goes out of scope right after the statement ends. 
Note the use as `LogMessage(arg1, arg2...)` 

If however it was used as `LogMessage variable(arg1,arg2)` then it would be printed when variable goes out of scope.

Reply:
Absolutely :)